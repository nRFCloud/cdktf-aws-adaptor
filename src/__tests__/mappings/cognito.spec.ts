import {
    CognitoIdentityPoolRolesAttachment,
} from "@cdktf/provider-aws/lib/cognito-identity-pool-roles-attachment/index.js";
import { CognitoIdentityPool } from "@cdktf/provider-aws/lib/cognito-identity-pool/index.js";
import { CognitoUserPoolClient } from "@cdktf/provider-aws/lib/cognito-user-pool-client/index.js";
import { CognitoUserPool } from "@cdktf/provider-aws/lib/cognito-user-pool/index.js";
import {
    CfnIdentityPool,
    CfnIdentityPoolRoleAttachment,
    CfnUserPool,
    CfnUserPoolClient,
} from "aws-cdk-lib/aws-cognito";
import { synthesizeElementAndTestStability } from "../helpers.js";

describe("Cognito mappings", () => {
    it("Should map AWS::Cognito::UserPoolClient", () => {
        synthesizeElementAndTestStability(
            CfnUserPoolClient,
            {
                enablePropagateAdditionalUserContextData: true,
                accessTokenValidity: 60,
                allowedOAuthFlows: ["code"],
                allowedOAuthFlowsUserPoolClient: true,
                analyticsConfiguration: {
                    externalId: "external-id",
                    applicationId: "application-id",
                    roleArn: "role-arn",
                    userDataShared: true,
                    applicationArn: "application-arn",
                },
                authSessionValidity: 60,
                callbackUrLs: ["callback-url"],
                clientName: "client-name",
                defaultRedirectUri: "default-redirect-uri",
                explicitAuthFlows: ["USER_PASSWORD_AUTH"],
                generateSecret: true,
                idTokenValidity: 60,
                logoutUrLs: ["logout-url"],
                preventUserExistenceErrors: "ENABLED",
                readAttributes: ["email"],
                refreshTokenValidity: 60,
                enableTokenRevocation: true,
                supportedIdentityProviders: ["COGNITO"],
                userPoolId: "user-pool-id",
                writeAttributes: ["email"],
                tokenValidityUnits: {
                    accessToken: "seconds",
                    idToken: "seconds",
                    refreshToken: "seconds",
                },
                allowedOAuthScopes: ["email"],
            },
            CognitoUserPoolClient,
            {
                name: "client-name",
                defaultRedirectUri: "default-redirect-uri",
                explicitAuthFlows: ["USER_PASSWORD_AUTH"],
                generateSecret: true,
                readAttributes: ["email"],
                supportedIdentityProviders: ["COGNITO"],
                userPoolId: "user-pool-id",
                writeAttributes: ["email"],
                allowedOauthFlows: ["code"],
                allowedOauthFlowsUserPoolClient: true,
                allowedOauthScopes: ["email"],
                analyticsConfiguration: [
                    {
                        externalId: "external-id",
                        applicationId: "application-id",
                        roleArn: "role-arn",
                        userDataShared: true,
                        applicationArn: "application-arn",
                    },
                ],
                callbackUrls: ["callback-url"],
                logoutUrls: ["logout-url"],
                preventUserExistenceErrors: "ENABLED",
                authSessionValidity: 60,
                idTokenValidity: 60,
                tokenValidityUnits: [
                    {
                        idToken: "seconds",
                        refreshToken: "seconds",
                        accessToken: "seconds",
                    },
                ],
                enableTokenRevocation: true,
                refreshTokenValidity: 60,
                accessTokenValidity: 60,
                enablePropagateAdditionalUserContextData: true,
            },
        );
    });

    it("Should map AWS::Cognito::IdentityPoolRoleAttachment", () => {
        synthesizeElementAndTestStability(
            CfnIdentityPoolRoleAttachment,
            {
                roles: {
                    authenticated: "authenticated-role",
                    unauthenticated: "unauthenticated-role",
                },
                roleMappings: {
                    "role-mapping": {
                        type: "Token",
                        ambiguousRoleResolution: "AuthenticatedRole",
                        identityProvider: "cognito-idp",
                        rulesConfiguration: {
                            rules: [{
                                claim: "claim",
                                matchType: "Equals",
                                roleArn: "role-arn",
                                value: "value",
                            }],
                        },
                    },
                },
                identityPoolId: "identity-pool-id",
            },
            CognitoIdentityPoolRolesAttachment,
            {
                roles: {
                    authenticated: "authenticated-role",
                    unauthenticated: "unauthenticated-role",
                },
                identityPoolId: "identity-pool-id",
                roleMapping: [
                    {
                        type: "Token",
                        ambiguousRoleResolution: "AuthenticatedRole",
                        identityProvider: "cognito-idp",
                        mappingRule: [
                            {
                                value: "value",
                                claim: "claim",
                                matchType: "Equals",
                                roleArn: "role-arn",
                            },
                        ],
                    },
                ],
            },
        );
    });

    it("Should map AWS::Cognito::IdentityPool", () => {
        synthesizeElementAndTestStability(
            CfnIdentityPool,
            {
                allowClassicFlow: true,
                allowUnauthenticatedIdentities: true,
                cognitoIdentityProviders: [
                    {
                        providerName: "provider-name",
                        serverSideTokenCheck: true,
                        clientId: "client-id",
                    },
                ],
                developerProviderName: "developer-provider-name",
                identityPoolName: "identity-pool-name",
                samlProviderArns: ["saml-provider-arn"],
                supportedLoginProviders: {
                    "login-provider": "login-provider",
                },
                openIdConnectProviderArns: ["open-id-connect-provider-arn"],
                identityPoolTags: [{ key: "hello", value: "hi" }, { key: "world", value: "universe" }],
                // FIXME: this is not supported by the provider yet
                // cognitoEvents: {
                //     "event": "event",
                // },
                // pushSync: {
                //     roleArn: "role-arn",
                //     applicationArns: ["application-arn"],
                // },
                // cognitoStreams: {
                //     roleArn: "role-arn",
                //     streamName: "stream-name",
                //     streamingStatus: "ENABLED",
                // },
            },
            CognitoIdentityPool,
            {
                allowClassicFlow: true,
                allowUnauthenticatedIdentities: true,
                cognitoIdentityProviders: [
                    {
                        providerName: "provider-name",
                        serverSideTokenCheck: true,
                        clientId: "client-id",
                    },
                ],
                developerProviderName: "developer-provider-name",
                identityPoolName: "identity-pool-name",
                samlProviderArns: ["saml-provider-arn"],
                supportedLoginProviders: {
                    "login-provider": "login-provider",
                },
                tags: {
                    hello: "hi",
                    world: "universe",
                },
            },
        );
    });

    it("Should map AWS::Cognito::UserPool", () => {
        synthesizeElementAndTestStability(
            CfnUserPool,
            {
                schema: [
                    {
                        name: "name",
                        mutable: true,
                        required: true,
                        attributeDataType: "String",
                        numberAttributeConstraints: {
                            maxValue: "max-value",
                            minValue: "min-value",
                        },
                        stringAttributeConstraints: {
                            maxLength: "max-length",
                            minLength: "min-length",
                        },
                    },
                ],
                enabledMfas: ["SMS"],
                adminCreateUserConfig: {
                    allowAdminCreateUserOnly: true,
                    inviteMessageTemplate: {
                        emailMessage: "email-message",
                        emailSubject: "email-subject",
                        smsMessage: "sms-message",
                    },
                    unusedAccountValidityDays: 60,
                },
                mfaConfiguration: "ON",
                userPoolAddOns: {
                    advancedSecurityMode: "OFF",
                },
                accountRecoverySetting: {
                    recoveryMechanisms: [
                        {
                            name: "name",
                            priority: 60,
                        },
                    ],
                },
                aliasAttributes: ["email"],
                autoVerifiedAttributes: ["email"],
                deviceConfiguration: {
                    challengeRequiredOnNewDevice: true,
                    deviceOnlyRememberedOnUserPrompt: true,
                },
                emailConfiguration: {
                    emailSendingAccount: "COGNITO_DEFAULT",
                    from: "from",
                    replyToEmailAddress: "reply-to-email-address",
                    configurationSet: "configuration-set",
                    sourceArn: "source-arn",
                },
                emailVerificationMessage: "email-verification-message",
                emailVerificationSubject: "email-verification-subject",
                lambdaConfig: {
                    createAuthChallenge: "create-auth-challenge",
                    customMessage: "custom-message",
                    defineAuthChallenge: "define-auth-challenge",
                    postAuthentication: "post-authentication",
                    postConfirmation: "post-confirmation",
                    preAuthentication: "pre-authentication",
                    preSignUp: "pre-sign-up",
                    kmsKeyId: "kms-key-id",
                    customSmsSender: {
                        lambdaArn: "lambda-arn",
                        lambdaVersion: "lambda-version",
                    },
                    customEmailSender: {
                        lambdaArn: "lambda-arn",
                        lambdaVersion: "lambda-version",
                    },
                    preTokenGeneration: "pre-token-generation",
                    userMigration: "user-migration",
                    verifyAuthChallengeResponse: "verify-auth-challenge-response",
                },
                usernameAttributes: ["email"],
                smsAuthenticationMessage: "sms-authentication-message",
                smsConfiguration: {
                    externalId: "external-id",
                    snsCallerArn: "sns-caller-arn",
                    snsRegion: "sns-region",
                },
                smsVerificationMessage: "sms-verification-message",
                policies: {
                    passwordPolicy: {
                        temporaryPasswordValidityDays: 60,
                        requireLowercase: true,
                        requireNumbers: true,
                        requireSymbols: true,
                        requireUppercase: true,
                        minimumLength: 60,
                    },
                },
                verificationMessageTemplate: {
                    defaultEmailOption: "CONFIRM_WITH_CODE",
                    smsMessage: "sms-message",
                    emailMessageByLink: "email-message-by-link",
                    emailSubjectByLink: "email-subject-by-link",
                    emailSubject: "email-subject",
                    emailMessage: "email-message",
                },
                userAttributeUpdateSettings: {
                    attributesRequireVerificationBeforeUpdate: ["email"],
                },
                deletionProtection: "true",
                usernameConfiguration: {
                    caseSensitive: true,
                },
                userPoolTags: {
                    "user-pool-tag": "user-pool-tag",
                },
                userPoolName: "user-pool-name",
            },
            CognitoUserPool,
            {
                name: "user-pool-name",
                usernameAttributes: ["email"],
                autoVerifiedAttributes: ["email"],
                aliasAttributes: ["email"],
                mfaConfiguration: "ON",
                smsAuthenticationMessage: "sms-authentication-message",
                smsConfiguration: {
                    externalId: "external-id",
                    snsCallerArn: "sns-caller-arn",
                    snsRegion: "sns-region",
                },
                softwareTokenMfaConfiguration: {
                    enabled: false,
                },
                emailVerificationMessage: undefined,
                emailVerificationSubject: undefined,
                emailConfiguration: {
                    emailSendingAccount: "COGNITO_DEFAULT",
                    configurationSet: "configuration-set",
                    replyToEmailAddress: "reply-to-email-address",
                    sourceArn: "source-arn",
                    fromEmailAddress: "from",
                },
                tags: {
                    "user-pool-tag": "user-pool-tag",
                },
                deletionProtection: "true",
                usernameConfiguration: {
                    caseSensitive: true,
                },
                userAttributeUpdateSettings: {
                    attributesRequireVerificationBeforeUpdate: ["email"],
                },
                passwordPolicy: {
                    minimumLength: 60,
                    requireLowercase: true,
                    requireNumbers: true,
                    requireSymbols: true,
                    requireUppercase: true,
                    temporaryPasswordValidityDays: 60,
                },
                smsVerificationMessage: undefined,
                verificationMessageTemplate: {
                    defaultEmailOption: "CONFIRM_WITH_CODE",
                    emailMessage: "email-message",
                    emailMessageByLink: "email-message-by-link",
                    emailSubject: "email-subject",
                    emailSubjectByLink: "email-subject-by-link",
                    smsMessage: "sms-message",
                },
                adminCreateUserConfig: {
                    allowAdminCreateUserOnly: true,
                    inviteMessageTemplate: {
                        emailMessage: "email-message",
                        emailSubject: "email-subject",
                        smsMessage: "sms-message",
                    },
                },
                accountRecoverySetting: {
                    recoveryMechanism: [
                        {
                            name: "name",
                            priority: 60,
                        },
                    ],
                },
                schema: [
                    {
                        name: "name",
                        mutable: true,
                        required: true,
                        attributeDataType: "String",
                        stringAttributeConstraints: {
                            maxLength: "max-length",
                            minLength: "min-length",
                        },
                        developerOnlyAttribute: false,
                        numberAttributeConstraints: {
                            maxValue: "max-value",
                            minValue: "min-value",
                        },
                    },
                ],
                deviceConfiguration: {
                    challengeRequiredOnNewDevice: true,
                    deviceOnlyRememberedOnUserPrompt: true,
                },
                lambdaConfig: {
                    createAuthChallenge: "create-auth-challenge",
                    customMessage: "custom-message",
                    defineAuthChallenge: "define-auth-challenge",
                    postAuthentication: "post-authentication",
                    postConfirmation: "post-confirmation",
                    preAuthentication: "pre-authentication",
                    preSignUp: "pre-sign-up",
                    kmsKeyId: "kms-key-id",
                    customSmsSender: {
                        lambdaArn: "lambda-arn",
                        lambdaVersion: "lambda-version",
                    },
                    preTokenGeneration: "pre-token-generation",
                    userMigration: "user-migration",
                    verifyAuthChallengeResponse: "verify-auth-challenge-response",
                    customEmailSender: {
                        lambdaArn: "lambda-arn",
                        lambdaVersion: "lambda-version",
                    },
                },
                userPoolAddOns: {
                    advancedSecurityMode: "OFF",
                },
            },
        );
    });
});
