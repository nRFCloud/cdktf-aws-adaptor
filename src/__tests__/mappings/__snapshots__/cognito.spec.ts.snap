// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cognito mappings > Should map AWS::Cognito::IdentityPool 1`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_cognito_identity_pool\\": {
      \\"resource_22C949BF\\": {
        \\"allow_classic_flow\\": true,
        \\"allow_unauthenticated_identities\\": true,
        \\"cognito_identity_providers\\": [
          {
            \\"client_id\\": \\"client-id\\",
            \\"provider_name\\": \\"provider-name\\",
            \\"server_side_token_check\\": true
          }
        ],
        \\"developer_provider_name\\": \\"developer-provider-name\\",
        \\"identity_pool_name\\": \\"identity-pool-name\\",
        \\"openid_connect_provider_arns\\": [
          \\"open-id-connect-provider-arn\\"
        ],
        \\"saml_provider_arns\\": [
          \\"saml-provider-arn\\"
        ],
        \\"supported_login_providers\\": {
          \\"login-provider\\": \\"login-provider\\"
        }
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.23.1\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\${join(\\\\\\"\\\\\\", [aws_cognito_identity_pool.resource_22C949BF.id, \\\\\\",\\\\\\", aws_cognito_identity_pool.resource_22C949BF.identity_pool_name])}\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;

exports[`Cognito mappings > Should map AWS::Cognito::IdentityPoolRoleAttachment 1`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_cognito_identity_pool_roles_attachment\\": {
      \\"resource_22C949BF\\": {
        \\"identity_pool_id\\": \\"identity-pool-id\\",
        \\"role_mapping\\": [
          {
            \\"ambiguous_role_resolution\\": \\"AuthenticatedRole\\",
            \\"identity_provider\\": \\"cognito-idp\\",
            \\"mapping_rule\\": [
              {
                \\"claim\\": \\"claim\\",
                \\"match_type\\": \\"Equals\\",
                \\"role_arn\\": \\"role-arn\\",
                \\"value\\": \\"value\\"
              }
            ],
            \\"type\\": \\"Token\\"
          }
        ],
        \\"roles\\": {
          \\"authenticated\\": \\"authenticated-role\\",
          \\"unauthenticated\\": \\"unauthenticated-role\\"
        }
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.23.1\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\${aws_cognito_identity_pool_roles_attachment.resource_22C949BF.id}\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;

exports[`Cognito mappings > Should map AWS::Cognito::UserPool 1`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_cognito_user_pool\\": {
      \\"resource_22C949BF\\": {
        \\"account_recovery_setting\\": {
          \\"recovery_mechanism\\": [
            {
              \\"name\\": \\"name\\",
              \\"priority\\": 60
            }
          ]
        },
        \\"admin_create_user_config\\": {
          \\"allow_admin_create_user_only\\": true,
          \\"invite_message_template\\": {
            \\"email_message\\": \\"email-message\\",
            \\"email_subject\\": \\"email-subject\\",
            \\"sms_message\\": \\"sms-message\\"
          }
        },
        \\"alias_attributes\\": [
          \\"email\\"
        ],
        \\"auto_verified_attributes\\": [
          \\"email\\"
        ],
        \\"deletion_protection\\": \\"true\\",
        \\"device_configuration\\": {
          \\"challenge_required_on_new_device\\": true,
          \\"device_only_remembered_on_user_prompt\\": true
        },
        \\"email_configuration\\": {
          \\"configuration_set\\": \\"configuration-set\\",
          \\"email_sending_account\\": \\"COGNITO_DEFAULT\\",
          \\"from_email_address\\": \\"from\\",
          \\"reply_to_email_address\\": \\"reply-to-email-address\\",
          \\"source_arn\\": \\"source-arn\\"
        },
        \\"lambda_config\\": {
          \\"create_auth_challenge\\": \\"create-auth-challenge\\",
          \\"custom_email_sender\\": {
            \\"lambda_arn\\": \\"lambda-arn\\",
            \\"lambda_version\\": \\"lambda-version\\"
          },
          \\"custom_message\\": \\"custom-message\\",
          \\"custom_sms_sender\\": {
            \\"lambda_arn\\": \\"lambda-arn\\",
            \\"lambda_version\\": \\"lambda-version\\"
          },
          \\"define_auth_challenge\\": \\"define-auth-challenge\\",
          \\"kms_key_id\\": \\"kms-key-id\\",
          \\"post_authentication\\": \\"post-authentication\\",
          \\"post_confirmation\\": \\"post-confirmation\\",
          \\"pre_authentication\\": \\"pre-authentication\\",
          \\"pre_sign_up\\": \\"pre-sign-up\\",
          \\"pre_token_generation\\": \\"pre-token-generation\\",
          \\"user_migration\\": \\"user-migration\\",
          \\"verify_auth_challenge_response\\": \\"verify-auth-challenge-response\\"
        },
        \\"mfa_configuration\\": \\"ON\\",
        \\"name\\": \\"user-pool-name\\",
        \\"password_policy\\": {
          \\"minimum_length\\": 60,
          \\"require_lowercase\\": true,
          \\"require_numbers\\": true,
          \\"require_symbols\\": true,
          \\"require_uppercase\\": true,
          \\"temporary_password_validity_days\\": 60
        },
        \\"schema\\": [
          {
            \\"attribute_data_type\\": \\"String\\",
            \\"developer_only_attribute\\": false,
            \\"mutable\\": true,
            \\"name\\": \\"name\\",
            \\"number_attribute_constraints\\": {
              \\"max_value\\": \\"max-value\\",
              \\"min_value\\": \\"min-value\\"
            },
            \\"required\\": true,
            \\"string_attribute_constraints\\": {
              \\"max_length\\": \\"max-length\\",
              \\"min_length\\": \\"min-length\\"
            }
          }
        ],
        \\"sms_authentication_message\\": \\"sms-authentication-message\\",
        \\"sms_configuration\\": {
          \\"external_id\\": \\"external-id\\",
          \\"sns_caller_arn\\": \\"sns-caller-arn\\",
          \\"sns_region\\": \\"sns-region\\"
        },
        \\"software_token_mfa_configuration\\": {
          \\"enabled\\": false
        },
        \\"tags\\": {
          \\"user-pool-tag\\": \\"user-pool-tag\\"
        },
        \\"user_attribute_update_settings\\": {
          \\"attributes_require_verification_before_update\\": [
            \\"email\\"
          ]
        },
        \\"user_pool_add_ons\\": {
          \\"advanced_security_mode\\": \\"OFF\\"
        },
        \\"username_attributes\\": [
          \\"email\\"
        ],
        \\"username_configuration\\": {
          \\"case_sensitive\\": true
        },
        \\"verification_message_template\\": {
          \\"default_email_option\\": \\"CONFIRM_WITH_CODE\\",
          \\"email_message\\": \\"email-message\\",
          \\"email_message_by_link\\": \\"email-message-by-link\\",
          \\"email_subject\\": \\"email-subject\\",
          \\"email_subject_by_link\\": \\"email-subject-by-link\\",
          \\"sms_message\\": \\"sms-message\\"
        }
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.23.1\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\${join(\\\\\\"\\\\\\", [aws_cognito_user_pool.resource_22C949BF.arn, \\\\\\",\\\\\\", aws_cognito_user_pool.resource_22C949BF.endpoint, \\\\\\",\\\\\\", \\\\\\"https://\${aws_cognito_user_pool.resource_22C949BF.endpoint}\\\\\\", \\\\\\",\\\\\\", aws_cognito_user_pool.resource_22C949BF.id])}\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;

exports[`Cognito mappings > Should map AWS::Cognito::UserPoolClient 1`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_cognito_user_pool_client\\": {
      \\"resource_22C949BF\\": {
        \\"access_token_validity\\": 60,
        \\"allowed_oauth_flows\\": [
          \\"code\\"
        ],
        \\"allowed_oauth_flows_user_pool_client\\": true,
        \\"allowed_oauth_scopes\\": [
          \\"email\\"
        ],
        \\"analytics_configuration\\": [
          {
            \\"application_arn\\": \\"application-arn\\",
            \\"application_id\\": \\"application-id\\",
            \\"external_id\\": \\"external-id\\",
            \\"role_arn\\": \\"role-arn\\",
            \\"user_data_shared\\": true
          }
        ],
        \\"auth_session_validity\\": 60,
        \\"callback_urls\\": [
          \\"callback-url\\"
        ],
        \\"default_redirect_uri\\": \\"default-redirect-uri\\",
        \\"enable_propagate_additional_user_context_data\\": true,
        \\"enable_token_revocation\\": true,
        \\"explicit_auth_flows\\": [
          \\"USER_PASSWORD_AUTH\\"
        ],
        \\"generate_secret\\": true,
        \\"id_token_validity\\": 60,
        \\"logout_urls\\": [
          \\"logout-url\\"
        ],
        \\"name\\": \\"client-name\\",
        \\"prevent_user_existence_errors\\": \\"ENABLED\\",
        \\"read_attributes\\": [
          \\"email\\"
        ],
        \\"refresh_token_validity\\": 60,
        \\"supported_identity_providers\\": [
          \\"COGNITO\\"
        ],
        \\"token_validity_units\\": [
          {
            \\"access_token\\": \\"seconds\\",
            \\"id_token\\": \\"seconds\\",
            \\"refresh_token\\": \\"seconds\\"
          }
        ],
        \\"user_pool_id\\": \\"user-pool-id\\",
        \\"write_attributes\\": [
          \\"email\\"
        ]
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.23.1\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\${join(\\\\\\"\\\\\\", [aws_cognito_user_pool_client.resource_22C949BF.id, \\\\\\",\\\\\\", aws_cognito_user_pool_client.resource_22C949BF.client_secret, \\\\\\",\\\\\\", aws_cognito_user_pool_client.resource_22C949BF.id, \\\\\\",\\\\\\", aws_cognito_user_pool_client.resource_22C949BF.name])}\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;
